### ISSUE TITLE
Flux temporarily broken on dev

### ISSUE THREAD

**Issue by jml** [weaveworks/flux#416]
I went to our dev Weave Cloud cluster and looked at the flux _for_ the dev cluster (proud-sky). 

When I clicked on a service, nothing happened. The history page remained as it was. At some point, I saw the error "Flux not connected. Check that it's running.” I don't recall what triggered that.

Eventually (many minutes later), it did start working. I went to deploy the service that brought me to using flux, and deployed it. It went through all the steps until it reached step 10, when it hung.

```
SERVICES

CURRENT RELEASES

CONFIGURE
Releasing monitoring/dev-grafana
Release job submitted, ID afc2dc6a-1d26-f1f2-0e28-a97bb196b2bd
Status: Release 1 service(s): monitoring/dev-grafana.

1) Queued.
2) Calculating release actions.
3) Release latest images to monitoring/dev-grafana
4) Clone the config repo.
5) Clone OK.
6) Update 2 images(s) in the resource definition file for monitoring/dev-grafana: grafana (quay.io/weaveworks/grafana:master-36ee41e -> quay.io/weaveworks/grafana:master-871271b), gfdatasource (quay.io/weaveworks/gfdatasource:master-1708580 -> quay.io/weaveworks/gfdatasource:master-da97417).
7) Update pod controller OK.
8) Commit and push the config repo.
9) Pushed commit: Release latest images to monitoring/dev-grafana
10) Release 1 service(s): monitoring/dev-grafana.
```

**Comment by rade** [comment]
This eventually showed up in our slack:
> (18:47:25) fluxy-dev: monitoring/dev-grafana: Release latest images to monitoring/dev-grafana. error: nats: timeout. failed

**Comment by ekimekim** [comment]
Still happening.
My actions:
* Attempted to search for a service, was able to type in search bar but no change in service list
* Attempted to cancel out of search bar, cross did not remove search term
* Attempted to click on a service, no effect
At this time I closed the page, so I don't know if it would've succeeded eventually.

**Comment by paulbellamy** [comment]
I'd suspect it is related to: https://github.com/weaveworks/service-conf/pull/540, wdyt @squaremo ?

**Comment by squaremo** [comment]
> I'd suspect it is related to: weaveworks/service-conf#540, wdyt @squaremo ?

They may be a bug in the NATS code, but I think it's more likely the introduction of timeouts is exposing some long-running operations.

In the case of applying releases (the last step above), there's a twenty minute timeout. That's a compromise between the API being synchronous, and not tying up resources indefinitely. Better would be if this step was done asynchronously, e.g., by starting it with an RPC, then checking on it.

**Comment by squaremo** [comment]
That's only the second part of the bug report, though. The first part seemed to be that API calls from the web UI aren't successful.

**Comment by rade** [comment]
> The first part seemed to be that API calls from the web UI aren't successful.

Or that the UI itself is misbehaving.

**Comment by squaremo** [comment]
#420 is probably causing some of the UI call failures

**Comment by awh** [comment]
Believed to be fixed by #427.
